geometry/TrianglesViewController.swift

```diff
@@ -6,64 +6,60 @@
 
 import UIKit
 
-class TrianglesViewController: UIViewController, TrianglesProtocol, TrianglesViewDataSource {
+class TrianglesViewController: UIViewController, SettingsViewControllerDelegate, TrianglesProtocol, TrianglesViewDataSource {
     private var triangles: Array<AnyObject> = []
-    
-//    private var redtvc = CGFloat() {didSet { updateUI()  } }
-//    private var greentvc = CGFloat() {didSet { updateUI()  } }
-//    private var bluetvc = CGFloat() {didSet { updateUI()  } }
-//    private var colors = [CGFloat]()
+    var redtvc: Float = 0.0
+    var greentvc: Float = 0.0
+    var bluetvc: Float = 0.0
     
     @IBOutlet weak var trianglesView: TrianglesView! {
         didSet {
             trianglesView.dataSourceColor = self
         }
     }
-    
     @IBAction func settingbutton(sender: AnyObject) {
-        self.view.setNeedsDisplay()
     }
-    @IBAction func loadTriangles(sender: UIButton) {
+    @IBAction func onLoad(sender: UIButton) {
+        loadTriangles()
+    }
+    
+    func loadTriangles() {
         if let path = NSBundle.mainBundle().pathForResource("triangles", ofType: "plist") {
             triangles = (NSArray(contentsOfFile: path) as? Array<AnyObject>)!
         }
-        self.view.setNeedsDisplay()
     }
-
+    
     func getTriangles() -> Array<AnyObject> {
         return triangles
     }
-    func getColor(sender: TrianglesView) -> [Float]? {
-        let settings = Settings.sharedInstance
-        return [settings.red,settings.green,settings.blue]
-    }
-    func updateUI() {
-        trianglesView?.setNeedsDisplay()
+    func fillSettingsViewControllerDelegate(rColor: Float, gColor: Float, bColor: Float) {
```

Comment:
Renamed to "setColor"

2016-06-06 22:25 GMT+06:00 psumarokov <notifications@github.com>:

> In geometry/TrianglesViewController.swift
> <https://github.com/todem5/geometry/pull/19#discussion_r65921812>:
>
> >      func getTriangles() -> Array<AnyObject> {
> >          return triangles
> >      }
> > -    func getColor(sender: TrianglesView) -> [Float]? {
> > -        let settings = Settings.sharedInstance
> > -        return [settings.red,settings.green,settings.blue]
> > -    }
> > -    func updateUI() {
> > -        trianglesView?.setNeedsDisplay()
> > +    func fillSettingsViewControllerDelegate(rColor: Float, gColor: Float, bColor: Float) {
>
> Rename to "setColor"
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/todem5/geometry/pull/19/files/fa72ce033a84f95207c9d7dbdd971e67f9f1d3e9#r65921812>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALUaDooo3_5DHmemfjeINAZzh0WMxsjHks5qJEnngaJpZM4Iu8NY>
> .
>



-- 
C Уважением Дмитрий!